# **Homelessness - Project 2**
### Analyzing the linear relationship between two quantitative variables.

# **Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that add extra features to the basic setup. The name of the modules is after the `import` statement, and the purpose is in a non-code comment after thew hashtag (#).



In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px
from IPython.display import Image
import statsmodels.api as sm


In [ ]:
# Replace 'image_url' with the URL of the image you want to display
image_url = 'https://endhomelessness.org/wp-content/uploads/2019/03/homelessness-statistics.jpg'

# Display the image
Image(url=image_url, width = 600)

# **Context**

The Continuum of Care (CoC) Program is designed to promote communitywide commitment to the goal of ending homelessness; provide funding for efforts by nonprofit providers, and State and local governments to quickly rehouse homeless individuals and families while minimizing the trauma and dislocation caused to homeless individuals, families, and communities by homelessness; promote access to and effect utilization of mainstream programs by homeless individuals and families; and optimize self-sufficiency among individuals and families experiencing homelessness. For more information on the Program, please visit https://www.hudexchange.info/programs/coc/

The U.S. Department of Housing and Urban Development (HUD) provides Point-in-Time (PIT) count reports of sheltered and unsheltered persons experiencing homelessness, by household type and subpopulation. This data is available at the national and state level, and for each CoC. HUD also provides Housing Inventory Count (HIC) reports, which provide a snapshot of a CoC’s inventory of beds and units available on the night designated for the count by program type, and include beds dedicated to serve persons who are homeless as well as persons in Permanent Supportive Housing. This raw data set contains PIT estimates of homelessness, and the corresponding accompanying HIC data from 2021.

Attribution: Adapted from U.S. Department of Housing and Urban Development


# **About the Dataset**

This dataset contains 100 rows corresponding to a random sample of localities (typically counties or similar large regions) that received Continuum of Care funding from HUD. A total of 21 variables are provided as listed in the table below.

| Variable Name(s)         | Description |
|:-------------------------|:--- |
| CoC Number<br>CoC Name | CoC locality identifier |
| CoC Category             | Community setting designation: Rural, Urban, etc |
| Type of Count            | Description of whether fully or partially sheltered individuals (or both) are included |
| Overall Homeless, 2021   | Total homeless count in all facilities indicated by the “Type of Count” variable |
| HMIS Participation Rate for Year-Round Beds (ES,TH,SH) | Proportion of facilities in the locality that participate in the Homeless Management Information System (HMIS) |
| Total Year-Round Beds (ES, TH, SH)<br>Total Year-Round Beds (ES)<br>Total Year-Round Beds (TH)<br>Total Year-Round Beds (SH) | (all types combined)<br>ES = Emergency Shelter<br>TH = Transitional Housing<br>SH = Safe Haven |
| Total Units for Households with Children (ES, TH, SH)<br>Total Beds for Households with Children (ES, TH, SH) | Number of units and beds designated for households with children |
| Sheltered ES Homeless 2021 | Estimate of the number of individuals sheltered in an Emergency Facility at the time of the study |
| Sheltered ES Homeless - Age 18 to 24, 2021 | Estimate of the number of individuals sheltered in an Emergency Facility by age |
| Sheltered ES Homeless - Female, 2021<br>Sheltered ES Homeless - Male, 2021<br>Sheltered ES Homeless - Trans+, 2021 | Estimate of the number of individuals sheltered in an Emergency Facility by gender |
| Sheltered ES Homeless - Hispanic/Latino, 2021<br>Sheltered ES Homeless - White, 2021<br>Sheltered ES Homeless - Black or African American, 2021<br>Sheltered ES Homeless - Asian or Pacific Islander<br>Sheltered ES Homeless - American Indian or Alaska Native, 2021<br>Sheltered ES Homeless - Multiple Races, 2021 | Estimate of the number of individuals sheltered in an Emergency Facility by race/ethnicity |



Let's take a look at the data. To do this, first we import it directly from the url below.



# **A Snippet of the Data**

In [ ]:
url='https://raw.githubusercontent.com/thamilton562/STAT108_Projects_Students/main/DataSets/Homelessness.csv'
df=pd.read_csv(url)

Next, we can display the data by *typing the name* of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [ ]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)
df

CoC Number                                           CoC Name  \
0      LA-505                     Monroe/Northeast Louisiana CoC   
1      MN-502                  Rochester/Southeast Minnesota CoC   
2      PA-508                     Scranton/Lackawanna County CoC   
3      OK-503                      Oklahoma Balance of State CoC   
4      OH-504                     Youngstown/Mahoning County CoC   
..        ...                                                ...   
95     KS-507                        Kansas Balance of State CoC   
96     CA-507                                   Marin County CoC   
97     TN-512  Morristown/Blount, Sevier, Campbell, Cocke Cou...   
98     MA-515                                     Fall River CoC   
99     CA-526  Amador, Calaveras, Mariposa, Tuolumne Counties...   

               CoC Category                   Type of Count  \
0         Largely Rural CoC            Sheltered-Only Count   
1         Largely Rural CoC            Sheltered-Only Count   
2      Largely Suburban CoC  Sheltered and full unsheltered   
3         Largely Rural CoC            Sheltered-Only Count   
4   Other Largely Urban CoC            Sheltered-Only Count   
..                      ...                             ...   
95        Largely Rural CoC            Sheltered-Only Count   
96     Largely Suburban CoC            Sheltered-Only Count   
97        Largely Rural CoC  Sheltered and full unsheltered   
98     Largely Suburban CoC  Sheltered and full unsheltered   
99        Largely Rural CoC            Sheltered-Only Count   

    Overall Homeless, 2021  \
0                       78   
1                      419   
2                      165   
3                      125   
4                       62   
..                     ...   
95                     783   
96                     225   
97                     472   
98                     324   
99                     151   

    HMIS Participation Rate for Year-Round Beds (ES, TH, SH)  \
0                                               0.233          
1                                               0.604          
2                                               0.839          
3                                               0.295          
4                                               0.192          
..                                                ...          
95                                              0.162          
96                                              0.727          
97                                              0.316          
98                                              0.877          
99                                              0.598          

    Total Year-Round Beds (ES, TH, SH)  Total Year-Round Beds (ES)  \
0                                  133                          69   
1                                  599                         378   
2                                  155                          64   
3                                  149                         135   
4                                  198                         187   
..                                 ...                         ...   
95                                1170                         703   
96                                 293                         163   
97                                 133                         117   
98                                 236                         212   
99                                 214                         156   

    Total Year-Round Beds (TH)  Total Year-Round Beds (SH)  \
0                           64                           0   
1                          221                           0   
2                           80                          11   
3                           14                           0   
4                           11                           0   
..                         ...                         ...   
95                      

# **INSTRUCTIONS**

* Use Python to analyze the data set and complete each of the following.
* Replace ellipsis (...) with the relavent names or code.  
* For problems that require a written response, double click the text box to start typing.
* Reference the 3 tutorials from activity for assistance.
* Attend office hours if you still need help.

This assignment is intended to explore the univariate linear relationships between quantitative variables in the data. Choose the dependent and independent variables based on your last name.

| Last Name | Dependent Variable (y) | Independent Variable (x) |
|-----------|------------------------|--------------------------|
| A-L       | Sheltered ES Homeless - Male, 2021 | Sheltered ES Homeless - Female, 2021 |
| M-Z       | Overall Homeless, 2021             | Total Year-Round Beds (ES, TH, SH) |




Based on your last name, which variable is explanatory and which is the response?

Explanatory: ...

Response: ...

What do you think the direction (positive/negative) and strength (weak/moderate/strong) will be?

# ...

# **QUESTION 1**

## Scatterplot

**1.1)** Construct a scatterplot that could be used to show the relationship between the variables stated above.

In [ ]:
# Create a scatter plot using Plotly Express

# Scatter plot for A-L
# STUDENTS: replace ... as stated
scatter_plot_AL = px.scatter(df,
                 # Replace ... with the explanatory variable name
                 x='Sheltered ES Homeless - Female, 2021',
                 # Replace ... with the response variable name
                 y='Sheltered ES Homeless - Male, 2021',
                         # Replace ... with a better x-axis label
                 labels={'Sheltered ES Homeless - Female, 2021': 'Number of Females in an Emergency Facility',
                         # Replace ... with a better y-axis label
                         'Sheltered ES Homeless - Male, 2021': 'Number of Males in an Emergency Facility'})

# Updating layout:
# STUDENTS: Do not change anything in this block of code.
scatter_plot_AL.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot A-L',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=575,                                 # Sets the width of the plot.
    height=400                                 # Sets the height of the plot for portrait mode.
)

# Show the plot:
scatter_plot_AL.show()

# Scatter plot for M-Z
# Create a scatter plot using Plotly Express
scatter_plot_MZ = px.scatter(df,
                 x='Total Year-Round Beds (ES, TH, SH)', #explanatory variable name
                 y='Overall Homeless, 2021', #response variable name
                 labels={'Total Year-Round Beds (ES, TH, SH)': 'Total Year-Round Beds (All Types)', #updates axis labels
                         'Overall Homeless, 2021': 'Total Homeless Count'})

# Updating layout:
scatter_plot_MZ.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot M-Z',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=575,                                 # Sets the width of the plot.
    height=400                                 # Sets the height of the plot for portrait mode.
)

# Show the plot
scatter_plot_MZ.show()

**1.2)** Describe the relationship between the two variables and include context, and identify the direction, strength, and form.

**A-L:** There is a strong positive linear relationship between the estimate of the number of females sheltered in an emergency facility and the estimate of the number of males sheltered in an emergency facility.

**M-Z:** There is a strong positive linear relationship between the total number of year-round beds and overall homelessness.

Is this what you expected?

Answers will vary.

# **Question 2**
## Correlation Coefficient

**2.1)** Calculate the value of the correlation coefficient.

In [ ]:
# Calculate the correlation coefficient

# STUDENTS: Replace ... as stated
# Replace the 1st ... with the response variable name.
# Replace the 2nd ... with the explanatory variable name
correlation_AL = df['Sheltered ES Homeless - Female, 2021'].corr(df['Sheltered ES Homeless - Male, 2021'])
correlation_MZ = df['Total Year-Round Beds (ES, TH, SH)'].corr(df['Overall Homeless, 2021'])

# Print the correlation coefficient
print(f"A-L: Correlation coefficient: r = {correlation_AL}")
print(f"\nM-Z: Correlation coefficient: r = {correlation_MZ}")

A-L: Correlation coefficient: r = 0.9742063256960228

M-Z: Correlation coefficient: r = 0.9825266368878709


**2.2)** Interpret the correlation coefficient.

**A-L:** There is a strong positive linear relationship between the estimate of the number of females sheltered in an emergency facility and the estimate of the number of males sheltered in an emergency facility.

**M-Z:** There is a strong positive linear relationship between the total number of year-round beds and overall homelessness.

# **Question 3**

## Least Squares Regression

**3.1)** Calculate the linear model and add the regression line to the scatterplot.


In [ ]:
# Calculate the LSRL and add the regression line to the scatterplot.

# A-L
# Extract the variables from the dataframe
# STUDENTS: Replace ... as stated below.
# Replace ... with the explantory variable name
X = df['Sheltered ES Homeless - Female, 2021']
# Replace ... with the response variable name
Y = df['Sheltered ES Homeless - Male, 2021']

# Add a constant to the explanatory variable for the regression model.
# This forces Statsmodels to calculate the y-intercept.
# STUDENTS: Do not change anything in this block of code.
X_with_const = sm.add_constant(X)

# Fit the linear regression model (calculate slope and y-intercept)
# STUDENTS: Do not change anything in this block of code.
model_AL = sm.OLS(Y, X_with_const).fit()

# STUDENTS: Replace ... as stated below.
scatter_plot_AL = px.scatter(df,
                 # Replace ... with the explanatory variable name
                 x='Sheltered ES Homeless - Female, 2021',
                 # Replace ... with the response variable name
                 y='Sheltered ES Homeless - Male, 2021',
                         # Replace ... with a better x-axis label
                 labels={'Sheltered ES Homeless - Female, 2021': 'Number of Females in an Emergency Facility',
                         # Replace ... with a better y-axis label
                         'Sheltered ES Homeless - Male, 2021': 'Number of Males in an Emergency Facility'})

# Updating layout
# STUDENTS: Do not change anything in this block of code.
scatter_plot_AL.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot A-L',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=600,                                 # Sets the width of the plot.
    height=400,                                # Sets the height of the plot for portrait mode.
    showlegend=False                           # Disables the legend (key) display
)

# Add the regression line to the scatterplot
scatter_plot_AL.add_scatter(x=X, y=model_AL.predict(X_with_const),
                         mode='lines', name='Regression Line')

# Show the plot
scatter_plot_AL.show()

# M-Z
# Extract the variables from the dataframe
X = df['Total Year-Round Beds (ES, TH, SH)']
Y = df['Overall Homeless, 2021']

# Add a constant to the explanatory variable for the regression model.
# This forces Statsmodels to calculate the y-intercept.
X_with_const = sm.add_constant(X)

# Fit the linear regression model (calculate slope and y-intercept)
model_MZ = sm.OLS(Y, X_with_const).fit()

# If we repeatedly run this section of code then a new regression line is
# added each run, unless we recreate the scatterplot each iteration.
scatter_plot_MZ = px.scatter(df,
                 x='Total Year-Round Beds (ES, TH, SH)', #explanatory variable name
                 y='Overall Homeless, 2021', #response variable name
                 labels={'Total Year-Round Beds (ES, TH, SH)': 'Total Year-Round Beds (All Types)', #updates axis labels
                         'Overall Homeless, 2021': 'Total Homeless Count'})

# Updating layout: Students do not change anything in this block of code.
scatter_plot_MZ.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot M-Z',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=600,                                 # Sets the width of the plot.
    height=400,                                # Sets the height of the plot for portrait mode.
    showlegend=False                           # Disables the legend (key) display
)

# Add the regression line to the scatterplot
scatter_plot_MZ.add_scatter(x=X, y=model_MZ.predict(X_with_const),
                         mode='lines', name='Regression Line')

# Show the plot
scatter_plot_MZ.show()

**3.2)** Print the coefficients (Intercept and Slope)



In [ ]:
# Print the coefficients (Intercept and Slope)

# STUDENTS: Do not change the code below.
# The coefficients were calculated when the regression line was added to the
# scatterplot and stored in 'model'. This assigns the values of the y-intercept
# and slope to 'coefficients', then prints the values.

# A-L
coefficients_AL = model_AL.params
coefficients_AL.name='Coefficients'
coefficients_AL

# M-Z
coefficients_MZ = model_MZ.params
coefficients_MZ.name='Coefficients'
coefficients_MZ

# Output the DataFrames
print("A-L Group:")
print(coefficients_AL)
print("\nM-Z Group:")
print(coefficients_MZ)

A-L Group:
const                                   52.647005
Sheltered ES Homeless - Female, 2021     1.100574
Name: Coefficients, dtype: float64

M-Z Group:
const                                 126.717550
Total Year-Round Beds (ES, TH, SH)      0.824899
Name: Coefficients, dtype: float64


**3.3)** Write the equation of the line that predicts waist circumference from BMI. To type 𝑦̂ you can type “y-hat” or “predicted y”. Do NOT round.



**A-L:** y-hat = 52.647005 + 1.100574x

**M-Z:** y-hat = 126.717550 + 0.824899x

# **Question 4**
## Slope
Interpret the slope


**A-L:** We expect the estimate of the number of males sheltered in an emergency facility to increase by 1.100574 for each 1 increase in the estimate of the number of females sheltered in an emergency facility. (No units required)

**M-Z:** We expect the overall homeless to increase by 0.824899 when the total number of year-round beds increases by 1. (no units required)


# **QUESTION 5**
## Y-Intercept

Interpret the y-intercept of the least squares regression line in the context of this study. State whether the interpretation is reasonable.

**A-L:** We expect the estimate of the number of males sheltered in an emergency facility to be 52.647005 when the estimate of the number of females sheltered in an emergency facility is 0. (No units required)

Yes, this makes sense. (No explanation required)

**M-Z:** We expect the overall homeless to 126.717550 when the total number of year-round beds is 1. (No units required)

Yes, this makes sense. (No explanation required)

# **Question 6**
## Prediction

Show work to predict the value of your dependent variable Los Angeles City & County CoC using the actual value of the independent variable shown in the table below. Round to two (2) decimal places.

| Last Name      | Independent Variable (x)                |
|----------------|----------------------------------------|
| **A-L**        | Sheltered ES Homeless - Female, 2021 = 6848 |
| **M-Z**        | Total Year-Round Beds (ES, TH, SH) = 21,107 |


**A-L:** y-hat = 52.64700531870045 + 1.1005741862776084(6848) = 7589.38

**M-Z:** y-hat = 126.71754978641798 + 0.8248988079686438(21107) = 17537.86

In [ ]:
# This is NOT in the student notebook.
# A-L prediction
intercept_AL = model_AL.params['const']
slope_AL = model_AL.params['Sheltered ES Homeless - Female, 2021']
predicted_AL = (intercept_AL + slope_AL * 6848)
print(f"A-L: y-hat = {intercept_AL} + {slope_AL}(6848) = {predicted_AL}")

# M-Z prediction
intercept_MZ = model_MZ.params['const']
slope_MZ = model_MZ.params['Total Year-Round Beds (ES, TH, SH)']
predicted_MZ = (intercept_MZ + slope_MZ * 21107)
print(f"M-Z: y-hat = {intercept_MZ} + {slope_MZ}(21107) = {predicted_MZ}")


A-L: y-hat = 52.64700531870045 + 1.1005741862776084(6848) = 7589.379032947762
M-Z: y-hat = 126.71754978641798 + 0.8248988079686438(21107) = 17537.856689580585


# **Question 7**
## Compare
**7.1)** Look up the actual value of your dependent variable for the value of the independent variable above. We know it is for Criminology majors, but be sure to type in the value of the independent variable listed in Question 6 in the code below.

In [ ]:
# Filter the datafram to find the row where your independent variable
#   equals the value from the table in Question 6.

#A-L
# STUDENTS:
# Replace the 1st ... with the explantory variable name
# Replace the 2nd ... with the value from the table in Question 6
# Replace the 3rd ... with the response variable name
actual_dependent_AL = df[df['Sheltered ES Homeless - Female, 2021']== 6848]['Sheltered ES Homeless - Male, 2021']
actual_AL = int(actual_dependent_AL.values[0])
print(f"A-L: The actual value of the dependent variable is: {actual_AL}")

#M-Z
actual_dependent_MZ = df[df['Total Year-Round Beds (ES, TH, SH)']== 21107]['Overall Homeless, 2021']
actual_MZ = int(actual_dependent_MZ.values[0])
print(f"M-Z: The actual value of the dependent variable is: {actual_MZ}")

A-L: The actual value of the dependent variable is: 7200
M-Z: The actual value of the dependent variable is: 17225


**7.2)** Compare the predicted value of your dependent variable from Question 6 to the actual value from 7.1. Use the actual name of your dependent variable, as modeled in activity.


**A-L:** The actual number of sheltered homeless males is lower than the predicted value.

**M-Z:** The actual total number of homeless peopls is lower than the predicted value.

# **QUESTION 8**

Generate a paragraph of at least 100 words to address one of the following questions. That is, answer only 8a or 8b, but not both.

**8a)** Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

...

--OR--

**8b)** Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?

...


<br><br>
### Once you are done and ready to submit, follow the instructions below to save as a PDF and submit to GradeScope.

### Save as PDF
Note 1: You do not have to select Print Preview. You can print directly from the notebook.
Note 2: Image and graph sizes have been set so you should be able to see them correctly without making any changes to the browser width or the layout (portrait vs landscape).
1. Run all code one last time and make sure your graphs can be seen.
2. File -> Print (or ctrl-p/cmnd-p)
3. Change the "Desination" to PDF.
4. Save the PDF, taking note of where it is saved.

### Submit to GradeScope
**Watch the "GradeScope Submission" video for help.**
1. Login to the Canvas course
2. Click on GradeScope in the course navigation.
3. If you see multiple courses in GradeScope, click on the STAT 108 course
4. Click on the name of the assignment that matches your data set
5. Click on "Submit Work", select PDF
6. Select the PDF you just created
7. You need to tell GradeScope which page each problem answer/output is on. You should see a list of problems on the left, and a display of pages (thumbnails) on the right. Assign pages to questions by clicking on the question number on the left, then clicking on all pages that question is on.
8. After ALL questions have been assigned to their respective page(s), click "Submit"

#### **Still need help? Your STAT 108 team is here to help. Take your laptop to office hours.**
